### PR TITLE
Bump rust msrv to 1.78

### PR DESCRIPTION
**Pull request by lucaato**

### CODE REVIEW

**Review by joshuachp — CHANGES_REQUESTED** [review]
Clap was pinned to a lower version for the MSRV bound, it can now be updated

**Review by joshuachp — APPROVED** [review]

**Review by rgallor — APPROVED** [review]

**Review by joshuachp — APPROVED** [review]

**Comment by rgallor** [comment]
Maybe you should also update the `astarte-device-sdk` and `astarte-message-hub-proto` in the Cargo.toml (at the moment you could use the git reference), since they have also been update to 1.78.0 MSRV

**Comment by lucaato** [comment]
We first need to merge the pr [on the sdk](https://github.com/astarte-platform/astarte-device-sdk-rust/pull/395) to be able to reference the master version of that crate too

**Review by harlem88 — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -66,16 +66,16 @@ serial_test = { workspace = true }
 tempfile = { workspace = true }
 
 [workspace.dependencies]
-astarte-device-sdk = "0.9.2"
+astarte-device-sdk = { git = "https://github.com/astarte-platform/astarte-device-sdk-rust", rev = "e8469c09100be2ad5446a7eb3bee359ee8314182"}
```
Comment (rgallor) [inline Cargo.toml:69]: you need to change also the sdk mock with the same info as the astarte-device-sdk

**Review by rgallor — CHANGES_REQUESTED** [review]

**Review by joshuachp — APPROVED** [review]

**Pull request merged**